tests/mesolve/open_system.py

```diff
@@ -45,10 +41,24 @@ def expect(self, t: float) -> Tensor:
     def expects(self, t: Tensor) -> Tensor:
         return torch.stack([self.expect(t_.item()) for t_ in t]).swapaxes(0, 1)
 
-    def loss(self, rho: Tensor) -> Tensor:
+    def loss_rho(self, rho: Tensor) -> Tensor:
         """Compute an example loss function from a given density matrix."""
         return dq.expect(self.loss_op, rho).real
 
+    def grads_loss_rho(self, t: float) -> Tensor:
+        """Compute the exact gradients of the example density matrix loss function with
+        respect to the system parameters."""
+        raise NotImplementedError
+
+    def losses_expect(self, expect: Tensor) -> Tensor:
+        """Compute example loss functions for each expectation values."""
+        return torch.stack(tuple(x.real for x in expect))
+
+    def grads_losses_expect(self, t: float) -> Tensor:
+        """Compute the exact gradients of the example expectation values loss functions
+        with respect to the system parameters."""
+        raise NotImplementedError
+
```

Comment:
How about the following naming:
`grads_loss_rho` -> `grads_loss`
`losses_expect` -> `stack_expect`
`grads_losses_expect` -> `grads_expect`
Might be a bit clearer, but not sure. It's not super important in any case.